feat: support aws credential and region overrides

rpocase opened the pull request:
the current implementation of awspub assumes a default profile is used. this adds support to override both the credentials and target region. overriding target region is to enable publication outside of the commercial partition.

rpocase committed:
refactor: add _get_client helper for boto3 clients

all calls for retrieving a boto3 client are currently
implemented inline. this adds a helper for a consistent
creation method in case there are config changes that
need to be made across all calls.

rpocase commented:
fyi @toabctl (i still need to follow up on getting you access)

toabctl commented on awspub/common.py:
```diff
@@ -69,3 +70,26 @@ def _get_regions(region_to_query: str, regions_allowlist: List[str]) -> List[str
+    those will be used to create the client
+    """
+
+    aws_access_key_id = os.getenv("AWS_ACCESS_KEY_ID")
```
those variables are already used by boto3 when provided. See https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html#environment-variables .

rpocase replied:
retested and that was definitely the problem. i thought i had tested that explicitly and it wasn't working before, but presumably i just did the access vars.

toabctl reviewed (commented):
is that for solving the problem with the partition sync where we have different creds?

rthill91 commented on awspub/common.py:
```diff
@@ -69,3 +70,26 @@ def _get_regions(region_to_query: str, regions_allowlist: List[str]) -> List[str
+        kwargs["aws_access_key_id"] = aws_access_key_id
+        kwargs["aws_secret_access_key"] = aws_secret_access_key
+    # interacting with partitions outside of commercial need
+    # to explicitly set the region_name
```
I think region name is always required, we just seldom need to specify because we use profiles that have a default region set already. As we switch to environment variables we'll need to start setting this for commercial partition as well

rpocase commented:
following comments from tom + ryan the change in functionality is a no-op. i've rebased and changed the initial commit, but happy to avoid merging. the benefit of keeping it would be if we want to play with a SESSION singleton (i just this initially but testing is always painful for global singletons)
